puppet/modules/kubernetes/manifests/apiserver.pp

```diff
@@ -48,8 +51,22 @@
   require ::kubernetes
 
   $_systemd_wants = $systemd_wants
-  $_systemd_requires = $systemd_requires
-  $_systemd_after = ['network.target'] + $systemd_after
+  if $aws_iam_authenticator_init {
+    $_systemd_after = ['aws-iam-authenticator-init.service'] + ['network.target'] + $systemd_after
+    $_systemd_requires = ['aws-iam-authenticator-init.service'] + $systemd_requires
+    if $auth_token_webhook_file == undef {
+      $_auth_token_webhook_file = '/etc/kubernetes/aws-iam-authenticator/kubeconfig.yaml'
+    } else {
+      $_auth_token_webhook_file = $auth_token_webhook_file
+    }
+    class{'kubernetes::aws_iam_authenticator_init':
```

Comment:
When I remove this option in the yaml, I would expect the service from this class to get cleaned up.
We should always call the class here and instead pass [`present`|`absent`] etc, like we are doing here for example 
https://github.com/jetstack/tarmak/blob/cd17222a19578dd82a7d46b67ec3a0dd03452f80/puppet/modules/vault_client/manifests/cert_service.pp#L35

Reply:
I've tried toggling it on and off on a "real" cluster and the logic seems to be sound :)